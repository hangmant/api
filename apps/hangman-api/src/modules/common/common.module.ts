import { Global, Module } from '@nestjs/common'
import { IDScalar } from './scalars/id.scalar'
import { GravatarService } from './services/gravatar.service'

@Global()
@Module({
  providers: [IDScalar, GravatarService],
  exports: [GravatarService]
})
export class CommonModule {}
