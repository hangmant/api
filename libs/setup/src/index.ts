export * from './helpers/create-microservices.helper';
export * from './helpers/create-service.helper';
