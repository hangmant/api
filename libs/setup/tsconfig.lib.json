{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "outDir": "../../dist/libs/setup"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "test", "**/*spec.ts"]
}
