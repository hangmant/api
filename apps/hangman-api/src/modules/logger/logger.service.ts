import { LoggerService as NestLoggerService } from '@nestjs/common'
import * as bunyan from 'bunyan'
import PrettyStream = require('bunyan-prettystream')

export class LoggerService implements NestLoggerService {
  private readonly _logger: bunyan

  constructor() {
    const packageJson = require('package.json')
    const prettyStream = new PrettyStream()

    prettyStream.pipe(process.stdout)

    this._logger = bunyan.createLogger({
      name: packageJson.name.toUpperCase(),
      streams: [
        {
          level: 'debug',
          type: 'raw',
          stream: prettyStream
        }
      ]
    })
  }

  log(message: any, ...params: Array<any>) {
    this._logger.info(message, params)
  }

  info(message: any, ...params: Array<any>) {
    this._logger.info(message, params)
  }

  error(message: string, trace?: string) {
    if (trace) {
      this._logger.error(message, trace)
    }
    this._logger.error(message)
  }

  warn(message: string, ...params: Array<any>) {
    this._logger.warn(message, params)
  }

  debug(message: string, ...params: any[]) {
    this._logger.info(message, params)
  }
}
