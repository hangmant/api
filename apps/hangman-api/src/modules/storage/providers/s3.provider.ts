import { Provider } from '@nestjs/common'
import { ConfigService } from '@nestjs/config'
import * as AWS from 'aws-sdk'
import { S3_PROVIDER } from '../storage.constants'

export const S3Provider: Provider = {
  provide: S3_PROVIDER,
  useFactory(configService: ConfigService) {
    const s3Config = configService.get('aws.s3')
    return new AWS.S3({
      accessKeyId: s3Config.accessKeyId,
      secretAccessKey: s3Config.secretAcessKey,
      signatureVersion: 'v4',
      region: 'us-east-1'
    })
  },
  inject: [ConfigService]
}
