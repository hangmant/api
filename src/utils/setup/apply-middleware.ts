import { ValidationPipe } from '@nestjs/common'
import { NestFastifyApplication } from '@nestjs/platform-fastify'
import * as BluebirdPromise from 'bluebird'
import helmet from 'helmet'
import * as passport from 'passport'

/** set custom promise to mongoose */
require('mongoose').Promise = BluebirdPromise

export async function applyMiddleware(app: NestFastifyApplication) {
  // await app.register(require('@fastify/express'))
 
  app.useGlobalPipes(new ValidationPipe())
  app.enableCors({
    origin: '*',
  })
  app.setGlobalPrefix('/api')
  app.use(helmet())

  // Passport
  app.use(passport.initialize())
  app.use(passport.session())
}
