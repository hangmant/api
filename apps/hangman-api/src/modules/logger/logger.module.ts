import { Module, Global } from '@nestjs/common'
import { loggerProviders } from './logger.providers'

@Global()
@Module({
  providers: [...loggerProviders],
  exports: [...loggerProviders]
})
export class LoggerModule {}
