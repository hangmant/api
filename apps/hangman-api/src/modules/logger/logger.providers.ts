import { Provider } from '@nestjs/common'
import { LoggerService } from './logger.service'

export const loggerProviders: Array<Provider> = [
  {
    provide: LoggerService,
    useValue: new LoggerService()
  }
]
