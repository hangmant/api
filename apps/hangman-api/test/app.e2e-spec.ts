import { HttpStatus } from '@nestjs/common'
import { FastifyAdapter } from '@nestjs/platform-fastify'
import { Test, TestingModule } from '@nestjs/testing'
import * as request from 'supertest'
import { AppModule } from '../src/app.module'
import { MongoMemoryServer } from 'mongodb-memory-server'
import { loggerServiceInstance } from '../src/modules/logger/logger.providers'
import { applyMiddleware } from '../src/utils/setup/apply-middleware'
import { MongoConfigService } from '../src/modules/mongo/mongoConfig.service'

describe.skip('AppController (e2e)', () => {
  let app
  let mongodb: MongoMemoryServer

  beforeAll(async () => {
    mongodb = new MongoMemoryServer()
    await mongodb.ensureInstance()

    jest.spyOn(MongoConfigService.prototype, 'mongoURI', 'get').mockReturnValue(await mongodb.getUri())

    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule]
    }).compile()

    app = moduleFixture.createNestApplication(new FastifyAdapter(), {
      logger: loggerServiceInstance
    })

    await applyMiddleware(app)

    await app.init()
  })

  afterAll(async () => {
    await app.close()
    await mongodb.stop()
  })

  describe('REST', () => {
    it('/ (GET)', async () => {
      const response = await request(app.getHttpServer()).get('/').expect(HttpStatus.NOT_FOUND)
      expect(response.body).toMatchSnapshot()
    })

    describe('Authentication /api/auth/login/jwt ', () => {
      it('should return not loggin when JWT is not given', async () => {
        /** TODO: Here error */
        const response = await request(app.getHttpServer()).post('/api/auth/login/jwt').send({ a: 1 })
        expect(true).toBe(true)
      })
    })
  })
})
