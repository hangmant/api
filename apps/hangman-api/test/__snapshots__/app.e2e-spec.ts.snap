// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) REST / (GET) 1`] = `
Object {
  "error": "Not Found",
  "message": "Not Found",
  "statusCode": 404,
}
`;
